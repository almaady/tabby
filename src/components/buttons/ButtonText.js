import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

/**
 * @class ButtonText - Is a button that only contains text
 */
class ButtonText extends React.PureComponent {
  /**
   * @property {string} text -The text of the button
   * @property {func} onClick - The function triggered by the button
   * @property {string} className - Just in case you need another class
   * @property {string} color -  The color of the button
   * @property {string} type - Button type
   * @property {bool} disabled - Button could be disabled
   * @property {string} borderColor - Button could have a border color
   * @see See './../../../styles/variables/_colors.scss' for the list of color names
   */
  static propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    color: PropTypes.string.isRequired,
    className: PropTypes.string,
    type: PropTypes.string,
    disabled: PropTypes.bool,
    borderColor: PropTypes.string
  };


  render() {
    const { color, onClick, text, className, type, disabled, borderColor } = this.props;
    return (
      <button
        className={classNames("btn", "btn--text", `bg--${color}`, `border--${borderColor}`,className)}
        onClick={onClick}
        type={type}
        disabled={disabled}
      >
        <p>{text}</p>
      </button>
    );
  }
}
ButtonText.defaultProps = {
  className: "",
  type: "button",
  disabled: false,
  borderColor: ""
};

export default ButtonText;
