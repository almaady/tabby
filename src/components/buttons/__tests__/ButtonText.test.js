import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";

import ButtonText from "../ButtonText";

describe("ButtonIcon", () => {
  const action = sinon.stub();
  let props;
  beforeEach(() => {
    props = {
      text: "text",
      onClick: action,
      color: "blue"
    };
  });
  it("should render correctly with default state", () => {
    const component = shallow(<ButtonText {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should call action when you click ", () => {
    const component = shallow(<ButtonText {...props} />);
    component.simulate("click");
    expect(action.called).toBeTruthy();
  });
  it("should render new className", () => {
    const component = shallow(<ButtonText {...props} className="btn-2" />);
    expect(component.hasClass("btn-2"));
  });
  it("should be a submit button", () => {
    const component = shallow(<ButtonText {...props} type="submit" /> );
    expect(component.find({type: "submit"})).toBeTruthy();
  });
});
