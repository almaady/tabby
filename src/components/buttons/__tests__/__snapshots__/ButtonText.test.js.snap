// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonText
    borderColor=""
    className=""
    color="blue"
    disabled={false}
    onClick={[Function]}
    text="text"
    type="button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        text
      </p>,
      "className": "btn btn--text bg--blue border--",
      "disabled": false,
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "text",
      },
      "ref": null,
      "rendered": "text",
      "type": "p",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          text
        </p>,
        "className": "btn btn--text bg--blue border--",
        "disabled": false,
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "text",
        },
        "ref": null,
        "rendered": "text",
        "type": "p",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
